common/persistence/client/quotas.go

```diff
@@ -169,6 +175,24 @@ func newPriorityRateLimiter(
 	)
 }
 
+func newPriorityDynamicRateLimiter(
+	rateFn quotas.RateFn,
+	requestPriorityFn quotas.RequestPriorityFn,
+	healthSignals p.HealthSignalAggregator,
+	dynamicParams DynamicRateLimitingParams,
+	logger log.Logger,
+) quotas.RequestRateLimiter {
+	rateLimiters := make(map[int]quotas.RequestRateLimiter)
+	for priority := range RequestPrioritiesOrdered {
+		rateLimiters[priority] = NewHealthRequestRateLimiterImpl(healthSignals, rateFn, dynamicParams, logger)
+	}
```

Comment:
not now, but I think we need to do some refactoring here. I see no reason why each priority needs to make its own decision on what the dynamic rate limit value should be.

Please add TODO here and we can sync re. the refactoring offline.